dom.go

```diff
@@ -2,9 +2,65 @@ package vecty
 
 import "github.com/gopherjs/gopherjs/js"
 
+// Renderable is a user-defined renderable component.
+type Renderable interface {
+	Component
+	Render() Component
+}
+
+// Core is the struct which all components embed.
+type Core struct {
+	component Renderable
+	body      Component
+}
+
+// Apply implements the Markup interface.
+func (c *Core) Apply(e *Element) {
+	e.AddChild(c.component)
+}
+
+// Unmount implements the Component interface.
+func (c *Core) Unmount() {
+	// Break the circular link between c (Core) and c.component (Renderable) or
+	// else one object could not be garbage collector without the other (both
+	// would have to be unused).
+	c.component = nil
+	c.body.Unmount()
+}
+
+// Node implements the Component interface.
+func (c *Core) Node() *js.Object {
+	return c.body.Node()
+}
+
+// Reconcile implements the Component interface.
+func (c *Core) Reconcile(oldComp Component) {
```

Comment:
You're 100% correct; and I think this may be the downfall of this PR. If I had to guess, you'll be strongly opposed to my last commit b947f2039e5fa771e3c4d56e2c248790f34ebb02 which fixes this. The alternative to this commit would be requiring users define the `Reconcile` boilerplate themselves.. but at that point the rest of this PR isn't super valuable (aside from Unmount, of course).

I'm also inherently opposed to this approach... just because `reflect` seems cheaty / nasty.. but I'd be interested to hear unbiased opinions? @shurcooL can give thoughts too maybe?